### ISSUE TITLE
[Bug]: RuntimeError: Check 'false' failed at C:\Jenkins\workspace\private-ci\ie\build-windows-vs2019\b\repos\openvino\src\inference\src\core.cpp:117: Failed to allocate 74784858112 bytes of memory

### ISSUE THREAD

**Issue by zhanghuayu-seu** [openvinotoolkit/anomalib#1224]
### Describe the bug

I successfully trained the PatchCore with my own dataset and save the .onnx weight, while it comes an error in the  inference procedure.

### Dataset

Other (please specify in the text field below)

### Model

PatchCore

### Steps to reproduce the behavior

Step1: Train the model: python tools/train.py --config custom_patchcore_v2.yaml

Step2: Inference the model: python tools/inference/openvino_inference.py --config custom_patchcore_2.yaml --weights results/patchcore/Corridors/run/weights/onnx/model.onnx --metadata results/patchcore/Corridors/run/weights/onnx/metadata.json --input image2.jpg --output results/111

### OS information

OS information:
- OS: [e.g. Windows 10]
- Python version: [e.g. 3.8.16]
- Anomalib version: [e.g. 0.6.0]
- PyTorch version: [e.g. 2.0.1]
- use CPU
- Any other relevant information: [e.g. I'm using a custom dataset]

### Expected behavior

no error comes

### Pip/GitHub

pip

### Configuration YAML

```yaml
dataset:
  name: Corridors
  format: folder
  path: ./datasets/Corridors
  normal_dir: normal
  abnormal_dir: null
  mask_dir: null #optional
  category: bottle
  task: classification
  normal_test_dir: null # optional
  extensions: null
  train_batch_size: 1
  eval_batch_size: 1
  inference_batch_size: 1
  num_workers: 8
  image_size: 256 # dimensions to which images are resized (mandatory)
  center_crop: null # dimensions to which images are center-cropped after resizing (optional)
  normalization: imagenet # data distribution to which the images will be normalized: [none, imagenet]
  transform_config:
    train: null
    eval: null
  test_split_mode: none # options: [from_dir, synthetic]
  test_split_ratio: 0.2 # fraction of train images held out testing (usage depends on test_split_mode)
  val_split_mode: synthetic # options: [same_as_test, from_test, synthetic]
  val_split_ratio: 0.5 # fraction of train/test images held out for validation (usage depends on val_split_mode)
  tiling:
    apply: false
    tile_size: 64
    stride: false
    remove_border_count: 0
    use_random_tiling: False
    random_tile_count: 16

model:
  name: patchcore
  backbone: wide_resnet50_2
  pre_trained: true
  layers:
    - layer2
    - layer3
  coreset_sampling_ratio: 0.1
  num_neighbors: 9
  normalization_method: min_max # options: [null, min_max, cdf]

metrics:
  image:
    - F1Score
    - AUROC
  threshold:
    method: adaptive #options: [adaptive, manual]
    manual_image: null

visualization:
  show_images: False # show images on the screen
  save_images: True # save images to the file system
  log_images: True # log images to the available loggers (if any)
  image_save_path: null # path to which images will be saved
  mode: full # options: ["full", "simple"]

project:
  seed: 42
  path: ./results

logging:
  logger: [] # options: [comet, tensorboard, wandb, csv] or combinations.
  log_graph: false # Logs the model graph to respective logger.

optimization:
  export_mode: openvino # options: torch, onnx, openvino

# PL Trainer Args. Don't add extra parameter here.
trainer:
  enable_checkpointing: true
  default_root_dir: null
  gradient_clip_val: 0
  gradient_clip_algorithm: norm
  num_nodes: 1
  devices: 1
  enable_progress_bar: true
  overfit_batches: 0.0
  track_grad_norm: -1
  check_val_every_n_epoch: 2 # Don't validate before extracting features.
  fast_dev_run: false
  accumulate_grad_batches: 1
  max_epochs: 4
  min_epochs: null
  max_steps: -1
  min_steps: null
  max_time: null
  limit_train_batches: 1.0
  limit_val_batches: 1.0
  limit_test_batches: 1.0
  limit_predict_batches: 1.0
  val_check_interval: 1.0 # Don't validate before extracting features.
  log_every_n_steps: 50
  accelerator: auto # <"cpu", "gpu", "tpu", "ipu", "hpu", "auto">
  strategy: null
  sync_batchnorm: false
  precision: 32
  enable_model_summary: true
  num_sanity_val_steps: 0
  profiler: null
  benchmark: false
  deterministic: false
  reload_dataloaders_every_n_epochs: 0
  auto_lr_find: false
  replace_sampler_ddp: true
  detect_anomaly: false
  auto_scale_batch_size: false
  plugins: null
  move_metrics_to_cpu: false
  multiple_trainloader_mode: max_size_cycle
```

### Logs

```shell
(Anomalib_py38) E:\anomalib-main>python tools/inference/openvino_inference.py --config custom_patchcore_2.yaml --weights results/patchcore/Corridors/run/weights/openvino/model.onnx --metadata results/patch
core/Corridors/run/weights/openvino/metadata.json --input image2.jpg --output results/111
(Anomalib_py38) E:\anomalib-main>python tools/inference/openvino_inference.py --config custom_patchcore_2.yaml --weights results/patchcore/Corridors/run/weights/openvino/model.onnx --metadata results/patch
core/Corridors/run/weights/openvino/metadata.json --input image2.jpg --output results/111
Traceback (most recent call last):
  File "tools/inference/openvino_inference.py", line 98, in <module>
    infer(args)
  File "tools/inference/openvino_inference.py", line 74, in infer
    inferencer = OpenVINOInferencer(path=args.weights, metadata=args.metadata, device=args.device)
  File "D:\anaconda3\envs\Anomalib_py38\lib\site-packages\anomalib\deploy\inferencers\openvino_inferencer.py", line 49, in __init__
    self.input_blob, self.output_blob, self.model = self.load_model(path)
  File "D:\anaconda3\envs\Anomalib_py38\lib\site-packages\anomalib\deploy\inferencers\openvino_inferencer.py", line 87, in load_model
    compile_model = ie_core.compile_model(model=model, device_name=self.device, config=self.config)
  File "D:\anaconda3\envs\Anomalib_py38\lib\site-packages\openvino\runtime\ie_api.py", line 398, in compile_model
    super().compile_model(model, device_name, {} if config is None else config),
RuntimeError: Check 'false' failed at C:\Jenkins\workspace\private-ci\ie\build-windows-vs2019\b\repos\openvino\src\inference\src\core.cpp:117:
Failed to allocate 74784858112 bytes of memory
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by blaz-r** [comment]
This might be a know issue, that is still present I believe, but with workaround possibly available in #967.

**Comment by samet-akcay** [comment]
Hi @zhanghuayu-seu, as @blaz-r pointed out, this seems to be a duplicate of #967. I'm closing this one, can you please refer to that issue? Thanks!
